File: valid_palindrome_rec.js
function validPalindrome (text) {

    // base case
    if (text.length <= 1){
        return true;
    }

    // recursive case
    if (text[0] == text[text.length-1]) {
        return validPalindrome(text.slice(1,text.length-1));
    }
    else{
        return false;
    }
}
console.log(validPalindrome('madam'));

File: valid_palindrome_rec.py
def is_palindrome_rec(text):

    # base case
    if len(text) <= 1:
        return True

    # recursive case
    if text[0] == text[-1]:
        return is_palindrome_rec(text[1:-1])
    else:
        return False
    
print(is_palindrome_rec('madam'))